src/esw/brushed_motors/Controller.cpp

```diff
@@ -197,12 +198,45 @@ void Controller::turnOn() const {
 }
 
 // REQUIRES: nothing
-// MODIFIES: isLive
+// MODIFIES: nothing
+// EFFECTS: Returns a combined ID for both the deviceAddress and motorID
+uint8_t Controller::combineDeviceMotorID() const {
+    return (deviceAddress << 3) | motorID;
+}
+
+// REQUIRES: nothing
+// MODIFIES: liveMap
 // EFFECTS: I2C bus, if not already live,
 // configures the physical controller.
 // Then makes live.
+
+std::unordered_map<uint8_t, LiveState> Controller::liveMap;
+
 void Controller::makeLive() {
-    if (isLive) {
+    // if (isLive) {
+    //     return;
+    // }
+
+    uint8_t key = combineDeviceMotorID();
+
+    // if key is absent, no motor with that key has been made live
+    auto it = liveMap.find(key);
+    if (it != liveMap.end()) {
```

Comment:
wait I think you'll want to put the unique lock outside of the if statement, I think it loses scope right after it leaves the if statement. So I think you can find a way to move it outside the if statement (the if statement is only necessary for emplacing the key,state into the liveMap